test/AutoDiff/SILOptimizer/differentiation_function_canonicalization.sil

```diff
@@ -45,9 +45,7 @@ bb0(%0 : $@differentiable @callee_guaranteed (Float) -> Float):
 // CHECK: bb0([[ARG:%.*]] : $@differentiable @callee_guaranteed (Float) -> Float):
 // CHECK:   [[ORIG_FN:%.*]] = differentiable_function_extract [original] [[ARG]]
 // CHECK:   [[JVP_FN:%.*]] = differentiable_function_extract [jvp] [[ARG]]
-// CHECK:   strong_retain [[JVP_FN]]
 // CHECK:   [[VJP_FN:%.*]] = differentiable_function_extract [vjp] [[ARG]]
-// CHECK:   strong_retain [[VJP_FN]]
```

Comment:
I think that these retains are actually necessary, because I'm seeing ASAN failures happening starting at this commit (https://bugs.swift.org/browse/SR-13973).

This commit stops emitting these changes because it makes the `.getOwnershipKind()` check at https://github.com/apple/swift/blob/d3d1f8d22d07233ae5714d810c26aff57066b453/lib/SILOptimizer/Mandatory/Differentiation.cpp#L503 & https://github.com/apple/swift/blob/d3d1f8d22d07233ae5714d810c26aff57066b453/lib/SILOptimizer/Mandatory/Differentiation.cpp#L1216 return None.

I don't really understand any of this, so I don't know what would be a good way to fix it. Is there some other check that we can do in Differentiation.cpp that does the right thing?

I did try completely deleting the check (so that Differentiation.cpp always emits the copy value operation), but that caused a lot of "Error! Found a leaked owned value that was never consumed." So I guess there are some cases where we do need a copy value and some cases where we don't need it, and I don't know the right way to distinguish between them :)

(There are also some other calls to `getOwnershipKind()` in Differentiation.cpp that might need to be updated, though I don't yet have any concrete examples of things that they break.)